src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let count = 0;
+
+  const adder = (...args) => {
+    if (args.length === 0) {
+      const sum = count;
+
+      count = 0;
+
+      return sum;
+    } else {
+      count += args.reduce((a, b) => a + b, 0);
```

Comment:
You are using `reduce` to sum the arguments, but since the function is supposed to receive a single argument at a time, you can directly add the argument to the `count`. Using `reduce` here is more complex than necessary and may lead to confusion about the intended usage of the `adder` function.